YOLayout/UIView+YOView.m

```diff
@@ -0,0 +1,119 @@
+//
+//  UIView+YOView.m
+//  YOLayoutExample
+//
+//  Created by Lucas Yan on 8/1/16.
+//  Copyright © 2016 YOLayout. All rights reserved.
+//
+
+#import "UIView+YOView.h"
+#import <objc/runtime.h>
+#import "YOLayout.h"
+#import "YOCGUtils.h"
+
+static char kAssociatedLayoutKey;
+
+@implementation UIView (YOView)
+
++ (void)initLayout {
+  [self exchangeImplementation:@selector(initWithFrame:) withImplementation:@selector(_initWithFrame:)];
+  [self exchangeImplementation:@selector(initWithCoder:) withImplementation:@selector(_initWithCoder:)];
+  [self exchangeImplementation:@selector(setFrame:) withImplementation:@selector(_setFrame:)];
+  [self exchangeImplementation:@selector(layoutSubviews) withImplementation:@selector(_layoutSubviews)];
+  [self exchangeImplementation:@selector(sizeThatFits:) withImplementation:@selector(_sizeThatFits:)];
+  [self exchangeImplementation:@selector(setNeedsLayout) withImplementation:@selector(_setNeedsLayout)];
+}
+
++ (void)exchangeImplementation:(SEL)originalImplementation withImplementation:(SEL)replacementImplementation {
+  Method originalMethod = class_getInstanceMethod(self, originalImplementation);
+  Method swizzledMethod = class_getInstanceMethod(self, replacementImplementation);
+  
+  BOOL didAddMethod =
+  class_addMethod(self,
+      originalImplementation,
+      method_getImplementation(swizzledMethod),
+      method_getTypeEncoding(swizzledMethod));
+  
+  if (didAddMethod) {
+    class_replaceMethod(self,
+      replacementImplementation,
+      method_getImplementation(originalMethod),
+      method_getTypeEncoding(originalMethod));
+  } else {
+    method_exchangeImplementations(originalMethod, swizzledMethod);
+  }
+}
+
+- (void)viewInit {
+  
+}
+
+- (CGSize)layoutWithLayout:(YOLayout *)layout size:(CGSize)size {
```

Comment:
Is this necessary? Why not just stick with the normal

    self.layout = [YOLayout layoutWithLayoutBlock:^CGSize(YOLayout *layout, CGSize size) {
    }];

The thing I like about the layout block, is that you are able to capture scope with the block so that you don't have to make properties for every single subview.